blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/ScheduleContextTest.java

```
@@ -385,4 +386,697 @@ public void testRecentlyScheduledShards() {
+
+        // precondition
+        Assert.assertEquals(1, ctx.getScheduledCount());
+
+        // when
+        SlotKey next = ctx.getNextScheduled();
+
+        // then
+        Assert.assertEquals(0, ctx.getScheduledCount());
+    }
+
+    @Test
+    public void testGetNextScheduledIncrementsRunningCount() {
+
+        // given
+        long now = 1234000L;
+        ScheduleContext ctx = new ScheduleContext(now, ringShards);
+        int shard = ringShards.get(0);
+        ctx.update(now-2, shard);
+        ctx.scheduleSlotsOlderThan(1);
+
+        // precondition
+        Assert.assertEquals(0, ctx.getRunningCount());
+
+        // when
+        SlotKey next = ctx.getNextScheduled();
+
+        // then
+        Assert.assertEquals(1, ctx.getRunningCount());
+    }
+
+    @Test
+    public void testGetNextScheduledReturnsTheSmallestGranularity() {
+
+        // given
+        long now = 1234000L;
+        long updateTime = now - 2;
+        int shard = ringShards.get(0);
+        Granularity gran = Granularity.MIN_5;
+        int slot = gran.slot(now);
+
+        ScheduleContext ctx = new ScheduleContext(now, ringShards);
+        ctx.update(updateTime, shard);
+        ctx.scheduleSlotsOlderThan(1);
+
+        // when
+        SlotKey next = ctx.getNextScheduled();
+
+        // then
+        Assert.assertEquals(shard, next.getShard());
+        Assert.assertEquals(slot, next.getSlot());
+        Assert.assertEquals(gran, next.getGranularity());
+    }
+
+    @Test
+    public void testGetNextScheduledChangesStateToRunning() {
+
+        // given
+        long now = 1234000L;
+        long updateTime = now - 2;
+        int shard = ringShards.get(0);
+        Granularity gran = Granularity.MIN_5;
+        int slot = gran.slot(updateTime);
+        Granularity coarserGran = null;
+        try {
+            coarserGran = gran.coarser();
+        } catch (GranularityException e) {
+            Assert.fail("Couldn't get the next coarser granularity");
```

Should we bubble up the exception so we have exception stack trace?